Rmd/introduction.Rmd

```diff
@@ -0,0 +1,165 @@
+This is the inverse of simulation, where we draw samples of data given parameters.
+
+Using _stan_ may seem a bit over the top for estimating bias from a few coin flips, but it will come in handy later when we look at more complicated problems.
+In order to conduct Bayesian inference, we will need to specify a [prior probability distribution](https://en.wikipedia.org/wiki/Prior_probability) for our parameter $p$ that reflects what we believe about the coin before we confront it with data.
+
+Let us assume we are completely ignorant, i.e. $p$ is equally likely to be any value between 0 and 1.
+In that case, we can use a _uniform prior_ bounded by 0 and 1, or in mathematical notation
+
+$$
+p \sim \mathcal{U}(0, 1)
+$$
+
+This is just another way of saying that any value of $p$ between 0 and 1 has the same _prior probability density_.
+
+When doing statistical inference we're often better off using distributions that don't have sharp edges like the uniform distribution.
+In our case here we could use, for example, a [Beta distribution](https://en.wikipedia.org/wiki/Beta_distribution) which is also bounded by 0 and 1 but tapers of smoothly towards the edges.
+
+_Stan_ is a so-called probabilistic programming language.
+This means that we write down the all the statistical relationships between variables in our model and then can use it to conduct inference on our desired parameters.
+
+In order to do our estimate of coin bias as above, we could write the following model in stan:
+
+```{r}
+coin_model <- '
+
+  data {
+    int trials;
+    int heads;
+  }
+
+  parameters {
+     real<lower = 0, upper = 1> prob_heads;
+  } 
+
+  model {
+    prob_heads ~ beta(5, 5);
+    heads ~ binomial(trials, prob_heads);
+  }
+'
+```
+
+In this model, we have defined three sections:
+- `data` represents all the things that aren't estimated, i.e. the things before the bar (|) in the likelihood function
+- `parameters` are the things that are estimated
+- `model` defines the prior and likelihood
+
+In order to conduct inference, we use the `sampling()` function in the `rstan` package
+
+```{r}
+library("rstan")
+model <- stan_model(model_code = coin_model) ## this may take a little while
+stan_result <- sampling(model, data = list(trials = 25, heads = k_heads))
+```
+
+We can inspect the results using
+
+```{r}
+stan_result
+```
+
+**Take 15 minutes*** to experiment with the code above: how does the error of the estimate change with the number of trials, or the given `k_heads`.
+Does the result match what you would expect?
+You could change the prior distribution for the bias to a uniform distribution by changing the `beta` distribution above to a `uniform(lower_bound, upper_bound)` (replacing `lower_bound` and `upper_bound` with the appropriate parameters). 
+Does this change your results?
+
+# Open-source tools
```

Comment:
Do we want to give some tools for interacting with stan models? i.e `tidybayes` and `ggdist`?

Reply:
I was thinking of this section as "open-source tools that achieve what we're trying to do here" - but yes, we'll probably want to introduce these at some point somewhere.